functions/record_transactions/record_transactions/auth.py

```
@@ -0,0 +1,100 @@
+
+    try:
+        jwks_url = f"https://cognito-idp.{aws_region}.amazonaws.com/{user_pool_id}/.well-known/jwks.json"
+        jwks_client = PyJWKClient(jwks_url)
+        try:
+            signing_key = jwks_client.get_signing_key_from_jwt(id_token)
+        except Exception as e:
+            logger.error(f"Failed to fetch or process JWKS: {str(e)}")
+            if "Failed to fetch jwks.json" in str(e):
+                raise AuthConfigurationError(
+                    "Auth configuration error: Failed to fetch or process JWKS"
+                )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)